### PR TITLE
fix: [CDS-83439]: fix for display of unsaved changes when ff CDS_GET_SERVICENOW_STANDARD_TEMPLATE is off in serviceNow create template

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
Bug - When `CDS_GET_SERVICENOW_STANDARD_TEMPLATE` is off, then while opening any old serviceNow create template, `useServiceNowTemplate` was getting replaced by `createType` and hence yaml changes and we see unsaved changes.

Solution: When FF is off then dont manipulate `useServiceNowTemplate`

Test cases - 

- when useServiceNowTemplate is true and CDS_GET_SERVICENOW_STANDARD_TEMPLATE ON

#### Screenshots
Before - 
image-20231103-125504

After - 

https://github.com/harness/harness-core-ui/assets/96036463/0fb3689f-8411-44d7-a144-23ca68fd3e5f

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger sonar

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
trigger messagecheck

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 1]
```diff
@@ -415,6 +416,10 @@ function FormContent({
     }
   }, [ticketValueType])
 
+  useEffect(() => {
```
Comment (sunnykesh-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:419]: I think ServiceNowCreateStepMode.tsx may be in user landing view so FF value will never change for this view . So, it should not be used with useEffect and set while initial value is calculated or set itself. cc: @pankajsingh-harness @chetanvekariya-harness

Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:419]: fixed

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Review by sunnykesh-harness — APPROVED** [review]
@satyam-harness Plz create a refactoring ticket to remove useEffect added now.

**Comment by satyam-harness** [comment]
retrigger eslint

**Comment by satyam-harness** [comment]
retrigger eslint

**Pull request merged**